src/transformers/models/time_series_transformer/configuration_time_series_transformer.py

```diff
@@ -0,0 +1,153 @@
+# coding=utf-8
+# Copyright 2022 kashif and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" TimeSeriesTransformer model configuration """
+from typing import List, Optional
+
+from gluonts.time_feature import get_lags_for_frequency, time_features_from_frequency_str
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+TIME_SERIES_TRANSFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "huggingface/tst-ett": "https://huggingface.co/huggingface/tst-ett/resolve/main/config.json",
+    # See all TimeSeriesTransformer models at https://huggingface.co/models?filter=time_series_transformer
+}
+
+
+class TimeSeriesTransformerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`~TimeSeriesTransformerModel`].
+    It is used to instantiate an TimeSeriesTransformer model according to the specified arguments, defining the model
+    architecture. Instantiating a configuration with the defaults will yield a similar configuration to that of
+    the TimeSeriesTransformer [huggingface/tst-ett](https://huggingface.co/huggingface/tst-ett) architecture.
+
+    Configuration objects inherit from  [`PretrainedConfig`] and can be used
+    to control the model outputs. Read the documentation from  [`PretrainedConfig`]
+    for more information.
+
+
+    Args:
+        prediction_length (`int`):
+            The prediction horizon for the model.
+        context_length (`int`, *optional*, default to `None`):
```

Comment:
```suggestion
        context_length (`int`, *optional*):
```
I think we don't include "default to None", in case it defaults to None :D